### DISCUSSION TITLE
Welcome to PicoCNC Discussions!

### DISCUSSION THREAD

**Discussion by phil-barrett** [phil-barrett/PicoCNC#1] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Suggest new features.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by gazzas23** [comment]
hi very new to github and to grblhal from what i can deduce you guys have done a fantastic jobon the picocnc board, however its a little more than my need for a simple 4 axis foam cuttter for rc wings, im looking to harness the power of grbl hal and the pico w to drive what would otherwise be a arduino cnc shield v3
 how do i get the grblhal web builder to delivee this  pinout?

![105925371-dda26b00-5ff4-11eb-8465-9f80e4bcfc91](https://github.com/phil-barrett/PicoCNC/assets/138290083/8aed098d-700a-4d96-8245-65db8d78cf07)
!

once confirmed i think i should be simple to connect the hardware to drive this [Screenshot_20240417_181001_YouTube](https://github.com/phil-barrett/PicoCNC/assets/138290083/739223df-93ea-4dee-a674-cefe2630580a)

appreciate this may have been asked before 

with best regards

gareth

**Reply by phil-barrett** [reply]
I'm not a big fan of those arduino shields - the lack of isolation can be an EMI problem. Also, I have no idea how you would retrofit a Pico onto that.

You can look at the map files from the grblHAL Web Builder to see the pinouts for various Pico/RP2040 versions.

**Reply by gazzas23** [reply]
its very much a diy challange but appreciate its a crude solution

i started with the generic_map_4axis.h file, i can see the direction gpio pin outs but not the step pins,  again its my lack of understanding thats at fault

thanks again for coming back to me

**Comment by JustinTimeToo** [comment]
I have the YD-RP2040. Is the image in the first post the pinout for the picocnc? I looked at the map file in the Web Builder but I can't make real heads and tail out of it. I searched for a picture like this but didn't get any results except this discussion.

**Reply by phil-barrett** [reply]
First, just compare the board to the actual pico, side by side. That should tell you if they are the same.

A lot depends on which board (map file) you chose.  PicoCNC uses shift registers so it will definitely be confusing if you don't know that.  Try Generic for a 3 axis setup.

**Reply by JustinTimeToo** [reply]
Thanks to the prompt reply, Phil!

While the YD-RP2040 is not quite the same as the Pico, which is a 2020, the pinouts are the same (the YD-RP2040 is newer). I'll try it and well see if the YD-RP2040 correspond to the picture above.
